### DISCUSSION TITLE
TypeScript code is not recogonized

### DISCUSSION THREAD

**Discussion by xsjcTony** [sveltejs/kit#8810] [Q&A] [answered]
### SvelteKit is not recognising TypeScript code.

This is a mini project with some modifications based on the `SvelteKit` official template... I added `tailwindcss` and settings for normal SPA (because I need the router so I used kit). 

Can anyone have a look at my reproduction repo to see what I did wrong? Thanks for the help. I believe it should be my own issue since I believe there's no way that Svelte won't work with TypeScript.

![Snipaste_2023-01-31_18-26-21](https://user-images.githubusercontent.com/45784210/215693924-761aecee-2033-4732-8fab-0c2492e13891.png)

### Reproduction

https://github.com/xsjcTony/50projects50days-svelte

### System Info

```Shell
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i9-11900K @ 3.50GHz
    Memory: 42.50 GB / 63.84 GB
  Binaries:
    Node: 18.12.1 - C:\NodeJS\node.EXE
    Yarn: 1.22.19 - C:\NodeJS\yarn.CMD
    npm: 9.4.0 - C:\NodeJS\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (109.0.1518.69)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @sveltejs/adapter-static: ^1.0.5 => 1.0.5
    @sveltejs/kit: ^1.3.6 => 1.3.6
    svelte: ^3.55.1 => 3.55.1
    vite: ^4.0.4 => 4.0.4
```

### Severity

blocking all usage of SvelteKit

**Comment by kristjanmar** [comment]
Just a guess, but the './types' path might be reserved because Svelte uses it for automatically typing the data returned from load functions.

Does it work if you change the name of the types file to something else?

**Comment by dummdidumm** [comment, marked as answer]
The reserved name is `./$types`, so that can't be it.

I believe it's the `svelte.config.ts` - you need that to be a `svelte.config.js`, there's no TS support for that config file.

Transfering this to a discussion.

**Reply by teemingc** [reply]
> I believe it's the `svelte.config.ts` - you need that to be a `svelte.config.js`, there's no TS support for that config file.

This sums up the issue. Without the valid svelte config file, it doesn't know it should use `vitePreprocess()` and thus no TypeScript support.

I've found a few changes that you'd need to make to get it working.

`svelte.config.ts` -> `svelte.config.js`
```diff
import { fileURLToPath, URL } from 'url'
import adapter from '@sveltejs/adapter-static'
import { vitePreprocess } from '@sveltejs/kit/vite'
- import type { Config } from '@sveltejs/kit'

+ /** @type {import('@sveltejs/kit').Config} */
- const config: Config = {
+ const config = {
  preprocess: vitePreprocess(),

  kit: {
    adapter: adapter({
      fallback: 'error.html'
    }),
    prerender: {
      entries: []
    },
    alias: {
      '@': fileURLToPath(new URL('src', import.meta.url))
    },
    paths: {
-      base: import.meta.env.PROD ? '/50projects50days-svelte' : ''
+      base: import.meta.env?.PROD ? '/50projects50days-svelte' : ''
    }
  }
}


export default config
```

Example here:
https://stackblitz.com/edit/github-m8vega?file=svelte.config.js,tsconfig.json

**Reply by xsjcTony** [reply]
Thanks so much for the help. One more question, what's the reason to use optional chaining for env?

**Reply by teemingc** [reply]
When I tried to run the dev server, it spat out an error saying `import.meta.env` was undefined. I guess it's undefined sometimes (maybe it runs before vite?) and that prevented the server from starting.

**Reply by xsjcTony** [reply]
Oh... then it means I cannot assume it's going to work for prod build either..

**Reply by teemingc** [reply]
Yeah, I'm not sure how to get Vite stuff working in `svelte.config.js`.

One workaround is to use good old `process.env`.

`process.env.NODE_ENV === 'production' ? '/50projects50days-svelte' : '',`

and run `NODE_ENV=production tsc && vite build`

https://stackblitz.com/edit/github-m8vega?file=svelte.config.js
